lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -631,6 +646,47 @@ void swift::checkActorIsolation(const Expr *expr, const DeclContext *dc) {
             return { false, expr };
           }
         }
+
+        // Actor-isolated properties can't be passed inout to async functions.
+        // If the function being called is async, we run checkMemberReference
+        // on all of the inout parameters to ensure that they do not belong to
+        // an actor.
+        FunctionType *fnType = apply->getFn()->getType()->getAs<FunctionType>();
+        if (fnType && fnType->isAsync()) {
+          // Since we get different types for the arg list depending on
+          // whether the function is called with one parameter or multiple
+          // parameters, just collect the inout expressions so that we don't
+          // have extra special-casing
+          llvm::SmallVector<Expr *, 3> inoutArgs = extractInoutArgs(apply);
+
+          // We have a list of arguments that were passed inout to this
+          // function, check that they don't come from an actor class
+          for (Expr *arg : inoutArgs) {
+            // Strip off all but the last LookupExpr before the DeclRefExpr.
+            // If the Decl from that LookupExpr comes from the actor, it's a
+            // problem.
+            while (LookupExpr *lookupArg = dyn_cast<LookupExpr>(arg)) {
```

Comment:
We could try to move this lookup into `checkMemberReference`, or maybe update `findMemberReference` to use it. I don't see a huge benefit to adding the loop to `checkMemberReference` since we'd need to run this loop in the caller as well to get the right `ConcreteDeclRef` to the member to pass into `checkMemberReference.